app/Http/Requests/Settings/ChangeLoginRequest.php

```diff
@@ -2,96 +2,99 @@
 
 namespace App\Http\Requests\Settings;
 
-use App\Facades\AccessControl;
-use App\Http\Requests\Session\LoginRequest;
+use App\Auth\Authorization;
+use App\Http\Requests\BaseApiRequest;
+use App\Http\Requests\Contracts\HasPassword;
+use App\Http\Requests\Contracts\HasUsername;
 use App\Http\Requests\Traits\HasPasswordTrait;
 use App\Rules\PasswordRule;
 use App\Rules\UsernameRule;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Gate;
 
-class ChangeLoginRequest extends LoginRequest
+class ChangeLoginRequest extends BaseApiRequest implements HasPassword
 {
-	public const OLD_USERNAME_ATTRIBUTE = 'oldUsername';
+	use HasPasswordTrait;
+
 	public const OLD_PASSWORD_ATTRIBUTE = 'oldPassword';
 
-	protected ?string $oldUsername = null;
-	protected ?string $oldPassword = null;
+	protected string $oldPassword;
+	protected ?string $username = null;
 
 	/**
 	 * Determines if the user is authorized to make this request.
-	 *
-	 * TODO: This method need to be rewritten after {@link \App\Actions\Settings\Login::do()} has been refactored.
-	 *
-	 * Normally, the request to change a user's password should
-	 * only be authorized for admin or non-locked users.
-	 * However, at the moment the method
-	 * {@link \App\Actions\Settings\Login::do()} is a "god" method and serves
-	 * three totally different use-cases mixed into one method (see comment
-	 * there).
-	 * We cannot reliably determine if the request is authorized without
-	 * knowing which of the use-case applies and thus without repeating
-	 * most of the logic of {@link \App\Actions\Settings\Login::do()}.
-	 * Hence, we authorize this request unconditionally and assume that
-	 * {@link \App\Actions\Settings\Login::do()} enforces correct
-	 * authorization.
-	 *
-	 * @return bool always true
 	 */
 	public function authorize(): bool
 	{
-		/*return AccessControl::is_logged_in() && (
-			AccessControl::is_admin() ||
-			!AccessControl::user()->is_locked
-		);*/
-
-		return true;
+		return Auth::check() && (
+			Gate::check('admin') ||
+			!Auth::authenticate()->is_locked
+		);
```

Comment:
To make this a little bit more efficient (and get rid of the Laravel Gates) this could be rewritten to

```suggestion
		$user = Auth::user();

		return $user !== null && (
			$user->isAdmin() ||
			!$user->is_locked
		);
```

It also appears a bit inconsistent to use introduce a Laravel Gate for the admin check, but not for the lock check. Moreover I just realized that `Auth::authenticate` is not properly defined on the `Auth` facade and only happens to work accidentally, due to Laravels "fall-trough magic". I wonder why PHPStan does not complain. IMHO, this should be solved, I write more about that in the overall comment.

Reply:
Rewritten completely.